src/main/java/com/elasticpath/tools/smcupgrader/impl/GitClientImpl.java

```diff
@@ -0,0 +1,325 @@
+	}
+
+	@Override public List<DiffEntry> getDiff() {
+		try (Git git = new Git(repository)) {
+			return git.diff().call();
+		} catch (final GitAPIException e) {
+			throw new RuntimeException(e);
+		}
+	}
+
+	@Override public List<DiffEntry> getDiff(final Ref otherBranch) {
+		try (Git git = new Git(repository)) {
+
+			final AbstractTreeIterator newTreeParser = prepareTreeParser(otherBranch);
+
+			return git.diff().setNewTree(newTreeParser).call();
+		} catch (final GitAPIException e) {
+			throw new RuntimeException(e);
+		}
+	}
+
+	@Override public void stage(final String path) {
+		try (Git git = new Git(repository)) {
+			git.add().addFilepattern(path).call();
+		} catch (final GitAPIException e) {
+			throw new RuntimeException(e);
+		}
+	}
+
+	@Override public void unstage(final String path) {
+		try (Git git = new Git(repository)) {
+			git.reset().addPath(path).call();
+		} catch (final GitAPIException e) {
+			throw new RuntimeException(e);
+		}
+	}
+
+	@Override public void delete(final String path) {
+		try (Git git = new Git(repository)) {
+			git.rm().addFilepattern(path).call();
+		} catch (final GitAPIException e) {
+			throw new RuntimeException(e);
+		}
+	}
+
+	@Override public void writeBlobContents(final String hexSha, final OutputStream outputStream) {
+		final ObjectId objectId = ObjectId.fromString(hexSha);
+		try {
+			final ObjectLoader loader = repository.open(objectId);
+			loader.copyTo(outputStream);
+		} catch (final IOException e) {
+			throw new RuntimeException(e);
+		}
+	}
+
+	@Override public Optional<String> getContentHashOfPathAtCommit(final String path, final RevCommit commit) {
+		try {
+			RevTree tree = commit.getTree();
+
+			try (TreeWalk treeWalk = new TreeWalk(repository)) {
+				treeWalk.addTree(tree);
+				treeWalk.setRecursive(true);
+				treeWalk.setFilter(PathFilter.create(path));
+
+				if (!treeWalk.next()) {
+					throw new IOException("Path not found in commit: " + path);
+				}
+
+				ObjectId objectId = treeWalk.getObjectId(0);
+				ObjectLoader loader = repository.open(objectId);
+
+				MessageDigest digest = MessageDigest.getInstance("SHA-256");
+				byte[] hashBytes = digest.digest(loader.getBytes());
+
+				StringBuilder hexString = new StringBuilder();
+				for (byte b : hashBytes) {
+					hexString.append(String.format("%02x", b));
+				}
+
+				return Optional.of(hexString.toString());
+			}
+		} catch (final IOException | NoSuchAlgorithmException ex) {
+			return Optional.empty();
+		}
+	}
+
+	private AbstractTreeIterator prepareTreeParser(final Ref ref) {
+		try (RevWalk walk = new RevWalk(repository)) {
+			final RevCommit commit = walk.parseCommit(ref.getObjectId());
+			final RevTree tree = walk.parseTree(commit.getTree().getId());
+
+			final CanonicalTreeParser treeParser = new CanonicalTreeParser();
+			try (ObjectReader reader = repository.newObjectReader()) {
+				treeParser.reset(reader, tree.getId());
+			}
+
+			walk.dispose();
```

Comment:
I think you can remove the dispose() call and rely on the try-with-resources, no?